src/foam/nanos/boot/Boot.java

```diff
@@ -64,12 +64,22 @@ public Boot(String datadir) {
     ArraySink arr = (ArraySink) serviceDAO_.select(new ArraySink());
     List      l   = perfectList(arr.getArray());
 
+    // Hierarchically nspecs
+    var hnSpecs   = new ArrayList<NSpec>();
     for ( int i = 0 ; i < l.size() ; i++ ) {
       NSpec sp = (NSpec) l.get(i);
       if ( ! sp.getEnabled() ) {
         logger.info("Disabled", sp.getName());
         continue;
       }
+
+      // Skip hierarchically nspecs to be loaded later since the service would
```

Comment:
I don't think we need to skip. Just create sub-contexts as needed if missing.

Reply:
Creating sub-context here polluted the root context. eg, when adding sub sub context say "bar" to "foo" sub-context which is under root, the "bar" sub-sub context also appears root.